### DISCUSSION TITLE
Update to v7.2.1 unable to start/update containers

### DISCUSSION THREAD

**Discussion by malcolmrigg** [nextcloud/all-in-one#3372] [Questions] [answered]
### Steps to reproduce
1. Version 7.0.0 - stop all containers
3. update master container
4. Once updated to Version 7.2.1 - hit start and update containers

### Expected behavior 
Containers update and start

### Actual behavior 
Database container fails to start with error below. 

### Host OS 
Asustor NAS - ADM 4.2.3.RK91

#### Nextcloud AIO version 
7.0.0 -> 7.2.1

#### Current channel 
latest

#### Other valuable info 

mastercontainer log error:

"Message: Client error: `POST http://localhost/v1.41/containers/nextcloud-aio-database/start` resulted in a `400 Bad Request` response:
{&quot;message&quot;:&quot;failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process (truncated...)"

I believe this is as a result of the "init" container flag change made. I manually disabled this flag in the database container and it then started successfully (though obviously I couldn't use the AIO interface to start the containers as it resets the flag back to enabled).

**Comment by szaimen** [comment]
Hi, is it only the database container that does not start?

**Comment by malcolmrigg** [comment]
> Hi, is it only the database container that does not start?

Not sure - that's the only container that I can see the init flag set on. It looks like once the database container fails to start AIO doesn't try the others. 

The other containers do start ok when manually started, but AIO is still saying there are updates... so given that and the fact I can't see the init flag in their config, I don't think they've been updated yet.

**Comment by szaimen** [comment]
Hm... Honestly this needs to work as it is a default docker thing so peobably something in asustors docker implementation is broken? 

Also I cannot reproduce this on my test instance..

**Comment by malcolmrigg** [comment]
OK, so I've done a little testing to see if docker works with the --init flag set on the asustor. 

Running this command:
`sudo docker run --init --rm -it --entrypoint sh alpine:latest`
Drops me into a terminal successfully. 

Doing an `ls /sbin/` shows the `docker-init` binary is there, and running `ps` shows `/sbin/docker-init` running as PID 1.

The `docker-init` binary I believe is what is truncated from the 
"unable to start container process (truncated) "
error message. 

I'm not saying that it's not a problem with Asustor's docker package, however there doesn't seem to be anything obvious wrong when using the `--init` flag. Is there anything I can run/look at to try and pin down the problem more precisely?

**Comment by malcolmrigg** [comment]
The similar command:

`sudo docker run --init -it --rm --entrypoint sh nextcloud/aio-postgresql:latest`

run with the AIO database image returns this error:

`docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: eaccess /sbin/docker-init: permission denied: unknown.`

**Comment by szaimen** [comment]
I see. I dont think the init flag has something to do with your issue then.

The underlying issue is this:

> docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed

I found this

https://forum.openmediavault.org/index.php?thread/46494-docker-is-not-working/

And this

https://www.google.com/search?q=%22docker%3A%20Error%20response%20from%20daemon%3A%20failed%20to%20create%20shim%20task%3A%20OCI%20runtime%20create%20failed%3A%20runc%20create%20failed%22#ip=1

Moving to discussions for now since I cannot reproduce the issue on my test instance as I said.

**Reply by malcolmrigg** [reply]
Thanks, I think you're partially right - it's not directly the init flag, however it does seem to be something to do with how the permissions work with the init flag set (remember that it runs ok with the init flag _unset_). If I run this:

`sudo docker run --init -it --rm --entrypoint sh --user root nextcloud/aio-postgresql:latest`

with the override to run as the root user instead of the postgres user - then it works. 

Is there something we can do when setting permissions in the docker file to allow the postgres user to run the docker-init process?

**Reply by malcolmrigg** [reply]
I got this working as below - thanks for you help.

**Comment by malcolmrigg** [comment, marked as answer]
I've managed to get this working, so for anybody else running AIO on an Asustor NAS - ADM 4.2.3.RK91 / Docker Engine 23.0.6.r2 or other systems where the permissions may be locked down slightly more than normal:

If you get this permissions error when trying to start a container:

`docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: eaccess /sbin/docker-init: permission denied: unknown.`

Then you need to manually set the permissions for the docker-init binary in your docker installation. In ADM this will be in the `.@plugins/AppCentral/docker-ce/bin/` directory, `cd` to this directory then run `chmod +x docker-init` to allow users other than root to run docker-init.
